### PR TITLE
ON-10316 # Added `useLoadDataState` hook

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: docs/useLoadDataState.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+| Property     | Type            | Description                                                                         |
+| ------------ | --------------- | ----------------------------------------------------------------------------------- |
+| `state`      | `LoadDataState` | The current state.                                                                  |
+| `handleLoad` | `function`      | A function that runs the `onLoad` function again. This can be used to refresh data. |
+
+`useLoadDataState()` can be used like:
+
+```js
+import { useLoadDataState } from '@oneblink/apps-react'
+import fetchData from 'fetchData' // Must be a function which returns a promise of some data
```
Comment (mymattcarroll) [inline docs/useLoadDataState.md:28]: Can we change this to an example of a function that returns a promise with some state in it. Using `fetch` would be a good example.

Comment (Zaxist) [reply docs/useLoadDataState.md:28]: yep

Comment (mymattcarroll) [reply docs/useLoadDataState.md:28]:
```suggestion
const fetchData = async () => {
  const response = await fetch('https://example.com/data')
  return await response.json()
}
```

File: docs/useLoadDataState.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const [state, refresh] = useLoadDataState(fetchData)
+
+switch (state.status) {
+  case 'LOADING':
+    return <Loading />
+  case 'ERROR':
+    return <Error message={state.error} />
+  case 'SUCCESS':
+  // RENDER UI
+}
```
Comment (mymattcarroll) [inline docs/useLoadDataState.md:39]: Can we wrap this part in a component function, it is a little bit misleading.

**Comment by mymattcarroll** [comment]
Just need to add a link to the docs for the hooks here: `/docs/README.md`

**Pull request merged**